File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/request/RegisterUserRequest.java
package dev.starryeye.book_library.controller.user.request;

import dev.starryeye.book_library.application.command.command.RegisterUserCommand;
import jakarta.validation.constraints.NotBlank;

public record RegisterUserRequest(
        @NotBlank(message = "username must not be blank")
        String username,
        Integer age
) {

    public RegisterUserCommand toCommand() {
        return new RegisterUserCommand(username, age);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/request/RegisterUserRequest.kt
package dev.starryeye.book_library.controller.user.request

import dev.starryeye.book_library.application.user.command.command.RegisterUserCommand
import jakarta.validation.constraints.NotBlank

class RegisterUserRequest(
    @field:NotBlank
    val username: String?,
    val age: Int?,
)

fun RegisterUserRequest.toCommand(): RegisterUserCommand {
    return RegisterUserCommand(
        username = this.username!!,
        age = this.age
    )
}